aws_common/CMakeLists.txt

```diff
@@ -83,6 +83,12 @@ find_package(GTest QUIET)
 if(NOT GTEST_FOUND)
   message(STATUS "Could not find GTest. Not building unit tests.")
 else()
+  include_directories(/usr/include/gmock /usr/src/gmock)
+  add_library(libgmock /usr/src/gmock/src/gmock-all.cc)
```

Comment:
Even though this is a ROS-agnostic package, we do depend on `catkin`, which provides utilities such as `catkin_add_gtest`, `catkin_add_gmock` and already takes care of setting up and building gtest/gmock if necessary (along with some fairly complex logic around path finding and edge cases: https://github.com/ros/catkin/blob/b1622e2042b8fa01df2896ea12d375acf6443beb/cmake/test/gtest.cmake#L229)

I think it'd be better to use that if possible.

Reply:
I'd be fine with the current approach, I wouldn't want this issue to block us. Overall looks good - great job adding the tests!